### PR TITLE
TEII-2300: prevent immediate clear selection

### PR DESCRIPTION
**Pull request by SergiiVdovareize**
https://inspera.atlassian.net/browse/TEII-2300
https://github.com/inspera/test-player/pull/3106

Because of several rerenders and click events in some cases the tooltip appears and immidiately dissappears.
The idea is to prevent such disappearing by setting minimum time to unfocus a note item. Otherwise it's made not mannually by a human.

### CODE REVIEW

**Review by B612nian — COMMENTED** [review]
Do we need to reset `this._focusedTime` back to `null` at some point?

**Comment by sergeyvdovareize** [comment]
It's not needed. Just need to prevent that immidiate cases

File: src/annotator/guest.coffee [hunk 1]
```diff
@@ -566,6 +567,10 @@ module.exports = class Guest extends Delegator
     excludeRange.some((item) -> element.closest(item))
 
   _onClearSelection: () ->
+    clearTime = Date.now()
+    if this._focusedTime && (clearTime - this._focusedTime) < 100
+      return false;
```
Comment (B612nian) [inline src/annotator/guest.coffee:572]: Please add a comment explaining why the 100 millis threshold was needed and also mention the ticket for future reference.

**Review by B612nian — CHANGES_REQUESTED** [review]

File: src/annotator/guest.coffee [hunk 2]
```diff
@@ -567,6 +567,12 @@ module.exports = class Guest extends Delegator
     excludeRange.some((item) -> element.closest(item))
 
   _onClearSelection: () ->
+    # TEII-2300
+    # Because of several rerenders and click events in some cases the tooltip appears and immidiately dissappears
```
Comment (B612nian) [inline src/annotator/guest.coffee:571]: immediately

File: src/annotator/guest.coffee [hunk 3]
```diff
@@ -567,6 +567,12 @@ module.exports = class Guest extends Delegator
...
+    # The idea is to prevent such disappearing by setting minimum time to unfocus a note item
+    # Otherwise it's made not mannually by a human
```
Comment (B612nian) [inline src/annotator/guest.coffee:573]: manually

**Review by B612nian — CHANGES_REQUESTED** [review]

**Comment by B612nian** [comment]
Hallo, @HuiguangChu. I see you were involved in some annotation bugfixing in the past related to Grading. I wonder if you see any deal-breakers with the approach taken in this PR that would be incompatible with the Grading side.

**Review by B612nian — COMMENTED** [review]
Asking @HuiguangChu for help on the review.

**Comment by B612nian** [comment]
I have asked Swati to coordinate with the other team so that @HuiguangChu can spend 45min code-reviewing this PR in the next 3 weeks.

**Review by B612nian — APPROVED** [review]
Approving and booking a testing slot for it next week: https://inspera.atlassian.net/wiki/spaces/TESTC/pages/3813113857/MVP10+Testing+Plan+09+05+22-13+05+22

**Do not merge** until it reaches its testing turn.

**Pull request merged**